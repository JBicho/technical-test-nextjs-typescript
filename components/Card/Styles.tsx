import styled from 'styled-components';

export const StyledCard = styled.div`
  align-items: center;
  background: rgba(255, 255, 255, 0.1);
  border-radius: 12px;
  backdrop-filter: blur(10px);
  box-shadow: var(--global-box-shadow);
  padding: var(--global-rhythm);
  margin: var(--global-rhythm);
  width: 100%;
  max-width: var(--global-max-width);
  transition:
    transform 0.3s ease-in-out,
    box-shadow 0.3s ease-in-out;

  @media (max-width: 430px), (orientation: landscape) {
    margin-top: calc(2.7 * var(--global-rhythm));
  }
`;
