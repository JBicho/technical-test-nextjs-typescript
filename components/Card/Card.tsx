import { ReactNode } from 'react';
import { StyledCard } from './Styles';

interface CardProps {
  children: ReactNode;
}

export const Card = ({ children }: CardProps) => {
  return <StyledCard>{children}</StyledCard>;
};
