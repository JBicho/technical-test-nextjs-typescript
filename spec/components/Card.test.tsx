import { render, screen } from '@testing-library/react';
import { Card } from '../../components/Card/Card';
import '../setupTests';

describe('Test Card Component', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  const childContent = 'This is some content';
  const InlineChild = () => <p>{childContent}</p>;

  const setupComponent = () =>
    render(
      <Card>
        <InlineChild />
      </Card>
    );

  it('Renders correctly and matches the snapshot', () => {
    const { asFragment } = setupComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render the component with children', () => {
    setupComponent();

    expect(screen.getByText(childContent)).toBeInTheDocument();
  });
});
