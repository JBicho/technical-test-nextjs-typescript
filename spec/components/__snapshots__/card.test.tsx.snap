// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card Component Renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Styles__StyledCard-ystdse-0 jYfiiW"
  >
    <p>
      This is some content
    </p>
  </div>
</DocumentFragment>
`;
